feat(cache): add ECR support for layer caching in buildkit

argoyle opened the pull request:
Introduce configuration options for AWS ECR as a remote layer cache  
backend in buildkit builds. Update the documentation and code to  
support caching layer configuration, improve build speed, and allow  
for better management of build artifacts.

peter-svensson commented on pkg/config/config_test.go:
```diff
@@ -329,3 +329,162 @@ targets:
+		},
+		{
+			name:       "configured cache us-east-1",
+			cache:      &ECRCache{Url: "123456789.dkr.ecr.us-east-1.amazonaws.com/cache"},
```
`123456789` is not a "valid" AWS Account id, should be 12 digits right? 😊

argoyle replied:
Good point, luckily only in a test. Should we add some validation? Do we have that for registries?

peter-svensson reviewed (commented):
Didn't find a test for the case when someone tries to configure:

```yaml
cache:
  gitlab:
    url: ...
```
Not something we support but how will the error message look like?

argoyle committed:
feat(cache): add ECR support for layer caching in buildkit

Introduce configuration options for AWS ECR as a remote layer cache  
backend in buildkit builds. Update the documentation and code to  
support caching layer configuration, improve build speed, and allow  
for better management of build artifacts.
